flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -34,7 +35,8 @@
 /**
  * Context object with optional arguments for a Flink Scan.
  */
-class ScanContext implements Serializable {
+@Internal
+public class ScanContext implements Serializable {
```

Comment:
This is made @Internal public because `TestArrayPoolDataIteratorBatcherRowData` access it from a diff package.

Reply:
Can we move that test?

Reply:
We can't. Otherwise, we also need to non-public `ArrayPoolDataIteratorBatcherRowData` from `source.reader` sub package to the parent `source` level. We will also need to move many other classes and test classes from sub packages to parent `source` level.